akka-typed/src/main/scala/akka/typed/javadsl/Actor.scala

```diff
@@ -192,6 +192,22 @@ object Actor {
   def immutable[T](`type`: Class[T]): BehaviorBuilder[T] = BehaviorBuilder.create[T]
 
   /**
+   * Construct an actor behavior that can react to lifecycle signals only. After spawning this actor
+   * from another actor (or as theguardian of an [[akka.typed.ActorSystem]]) it will be executed
+   * within an [[ActorContext]] that allows access to the system, spawning and watching other
+   * actors, etc.
+   */
+  def signal[T](
+    init:     Procedure[ActorContext[T]],
+    onSignal: JapiFunction2[ActorContext[T], Signal, Behavior[T]]): Behavior[T] =
+    Behavior.DeferredBehavior { ctx ⇒
+      init.apply(ctx.asJava)
+      new BehaviorImpl.ImmutableBehavior[T](
+        (_, _) ⇒ Actor.empty,
+        { case (ctx, sig) ⇒ onSignal.apply(ctx.asJava, sig) })
```

Comment:
or:
```
    val emptyFun = new JapiFunction2[ActorContext[T], T, Behavior[T]] {
      override def apply(ctx: ActorContext[T], msg: T) = Actor.empty
    }
    immutable(emptyFun, signal)
```

Reply:
Thanks! I was trying that but failed at this empty Java function.

Reply:
Done